Source/santad/EventProviders/SNTDeviceManager.mm

```diff
@@ -0,0 +1,292 @@
+    return;
+  }
+
+  long mountMode = m->event.mount.statfs->f_flags;
+  pid_t pid = audit_token_to_pid(m->process->audit_token);
+  LOGI(@"SNTDeviceManager: mount syscall arriving from path: %s, pid: %d, fflags: %lu",
+       m->process->executable->path.data, pid, mountMode);
+
+  DADiskRef disk =
+    DADiskCreateFromBSDName(NULL, self.diskArbSession, m->event.mount.statfs->f_mntfromname);
+  CFAutorelease(disk);
+
+  // TODO(tnek): Log all of the other attributes available in diskInfo into a structured log format.
+  NSDictionary *diskInfo = CFBridgingRelease(DADiskCopyDescription(disk));
+  BOOL isRemovable = [diskInfo[(__bridge NSString *)kDADiskDescriptionMediaRemovableKey] boolValue];
+  BOOL isUSB = [diskInfo[@"DADeviceProtocol"] isEqualTo:@"USB"];
+
+  if (!isRemovable || !isUSB) {
+    es_respond_auth_result(self.client, m, ES_AUTH_RESULT_ALLOW, false);
+    return;
+  }
+
+  if (self.remountArgs != nil && [self.remountArgs count] > 0) {
+    long remountOpts = mountArgsToMask(self.remountArgs);
+    if (mountMode & remountOpts) {
+      es_respond_auth_result(self.client, m, ES_AUTH_RESULT_ALLOW, false);
+      return;
+    }
+
+    long newMode = mountMode | remountOpts;
+    LOGI(@"SNTDeviceManager: remounting device '%s'->'%s', flags (%lu) -> (%lu)",
+         m->event.mount.statfs->f_mntfromname, m->event.mount.statfs->f_mntonname, mountMode,
+         newMode);
+    [self remount:disk mountMode:newMode];
+  }
+  es_respond_auth_result(self.client, m, ES_AUTH_RESULT_DENY, false);
+}
+
+- (void)remount:(DADiskRef)disk mountMode:(long)remountMask {
+  NSArray<NSString *> *args = maskToMountArgs(remountMask);
+  CFStringRef *argv = (CFStringRef *)calloc(args.count + 1, sizeof(CFStringRef));
+  CFArrayGetValues((__bridge CFArrayRef)args, CFRangeMake(0, (CFIndex)args.count),
+                   (const void **)argv);
+
+  DADiskMountWithArguments(disk, NULL, kDADiskMountOptionDefault, diskMountedCallback,
+                           (__bridge void *)self, (CFStringRef *)argv);
+  free(argv);
+}
+
+- (void)handleESMessageWithTimeout:(const es_message_t *)m
+                        withClient:(es_client_t *)c
+                           timeout:(dispatch_time_t)timeout API_AVAILABLE(macos(10.15)) {
+  // ES will kill our whole client if we don't meet the es_message auth deadline, so we try to
+  // gracefully handle it with a deny-by-default in the worst-case before it can do that.
+  // This isn't an issue for notify events, so we're in no rush for those.
+  std::shared_ptr<std::atomic<bool>> responded;
+  if (m->action_type == ES_ACTION_TYPE_AUTH) {
+    responded = std::make_shared<std::atomic<bool>>(false);
+    dispatch_after(timeout, self.esAuthQueue, ^(void) {
+      if (responded->load()) return;
+      LOGE(@"SNTDeviceManager: deadline reached: deny pid=%d ret=%d",
+           audit_token_to_pid(m->process->audit_token),
+           es_respond_auth_result(c, m, ES_AUTH_RESULT_DENY, false));
```

Comment:
We have similar logic in `SNTEndpointSecurityManager.mm`. As a future TODO, it would be nice to consider having centralized logic for handling timeouts of messages from arbitrary clients

Reply:
Added it as to-do for now. This PR in this form actually originated from when I was trying to make the SNTEndpointSecurityManager's messageHandler actually handle all of the messages